Source/GTMSessionFetcher.m

```diff
@@ -2853,6 +2857,22 @@ - (void)URLSession:(NSURLSession *)session
   }];
 }
 
+#if GTM_SUPPORTS_METRICS_COLLECTION
+- (void)URLSession:(NSURLSession *)session
+              task:(NSURLSessionTask *)task
+didFinishCollectingMetrics:(NSURLSessionTaskMetrics *)metrics {
+  @synchronized(self) {
+    GTMSessionFetcherMetricsCollectionBlock metricsCollectionBlock = _metricsCollectionBlock;
+    if (metricsCollectionBlock) {
+      GTMSessionMonitorSynchronized(self);
```

Comment:
Move this GTMSessionMonitorSynchronized to immediately after the @synchronized.

Reply:
Done.